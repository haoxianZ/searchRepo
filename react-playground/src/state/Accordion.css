.Accordion {
    margin: 0;
    padding: 0;
    list-style-type: none;
  }
  
  .item {
    margin-bottom: 2px;
  }
  
  .item button {
    background: lightgrey;
    border: 1px solid #222;
    padding: 10px 20px;
    width: 100%;
    font-size: 18px;
    font-weight: bold;
  }
  
  .item p {
    background: #fdfcfc;
    border: 1px solid #777;
    border-top: none;
    margin: 0;
    padding: 20px 10px;
  }