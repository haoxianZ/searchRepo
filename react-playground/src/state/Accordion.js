import React from 'react';
import './Accordion.css'
class Accordion extends React.Component {
    static defaultProps = {sections:[]};
    state ={currentTabIndex : null}
    handleClick (index){
        console.log('clikc')
        this.setState({currentTabIndex :index})
    }
    renderButtons (){
        return this.props.sections.map(
            (section,index)=>(<li className='item' key={index}><button 
            onClick={() => this.handleClick(index)}
            key={index}>
                {section.title}
            </button>
            {this.renderContent(index)}</li>))
            };
    renderContent (index) {
        const currentSection = this.props.sections[this.state.currentTabIndex];
        if( index === this.state.currentTabIndex){
            return (
            <p className='content'>
                {currentSection.content}
            </p>
        )
        }
        
    }
    render(){
        //{this,props.tabs.length need to be there else error}
        return (<div>
                <ul>
                    {this.renderButtons()}
                </ul>
        
            </div>)
    }
}

export default Accordion;