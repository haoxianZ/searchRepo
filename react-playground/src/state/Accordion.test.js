import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';
const sectionsProp = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]
describe('Accordion component', ()=>{
    it('render without crashing empty',()=>{
        const wrapper = shallow(<Accordion/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('render no section by default', ()=>{
        const wrapper = shallow(<Accordion sections={sectionsProp}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('only open one section a time',()=>{
        const wrapper = shallow(<Accordion sections={sectionsProp}/>)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('open when click',()=>{
        const wrapper = shallow(<Accordion sections={sectionsProp}/>)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})