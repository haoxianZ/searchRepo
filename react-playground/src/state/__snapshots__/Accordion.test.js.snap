// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component only open one section a time 1`] = `
<div>
  <ul>
    <li
      className="item"
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="item"
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      className="item"
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
      <p
        className="content"
      >
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion component open when click 1`] = `
<div>
  <ul>
    <li
      className="item"
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="item"
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p
        className="content"
      >
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      className="item"
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion component render no section by default 1`] = `
<div>
  <ul>
    <li
      className="item"
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="item"
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      className="item"
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion component render without crashing empty 1`] = `
<div>
  <ul />
</div>
`;
